queue/ArrayQueue.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Arrays;
+/**
+ * An array implementation of a queue
+ * 
+ * @author Harry Willis
+ */
+
+public class ArrayQueue<T> implements Queue<T>{
+    
+    private T[] queue;
+
+    private int front;
+    
+    // the next available spot in the queue
+    private int back;
+
+    private int size;
+
+    
+    /**
+     * Build default ArrayQueue with size 5
+     */
+    public ArrayQueue(){
+        this(5);
+    }
+
+    /**
+     * Build ArrayQueue with specified size
+     */
+    @SuppressWarnings("unchecked")
+    public ArrayQueue(int size){
+        this.front = 0;
+        
+        this.back = 0;
+
+        this.size = size;
+
+        this.queue = (T[]) new Object[size];
+    }
+    /**
+     * enqueue(T item)
+     * add an element to the back of the queue, extending capacity if necessary
+     * @param item to be added
+     * @return item added
+     */
+    public T enqueue(T item){
+        if(this.back + 1 == this.size){
```

Comment:
Similar to my stylistic comment above, generally ifs and whiles and fors etc will be written with spaces between their components for legibility.

`if (this.back + 1 == this.size) {`

Again, you've been consistent throughout the code and that's key and if you prefer how it is now, feel free to leave it.